src/msocket_server.c

```diff
@@ -71,19 +71,18 @@ void msocket_server_destroy(msocket_server_t *self){
 #endif
 
    if(self != 0){
+      self->cleanupStop = 1;
```

Comment:
I'd rather be safe than sorry and protect this using a lock or mutex. On some architectures this could become 2 instructions.
Set immediate (some register) to value 1
Store (that register) at the DRAM address where self->cleanupStop is located.

Is that always safe without lock in all situations?

Reply:
With the data type being small enougth to be atomic and the producer not reading the value before changing it I think it is safe without protection.

The reader will always read either 0 or 1 and if two producers try to change it at the same time it is undetermined who will get its change made in the long term regardless if the both take the sem or not.